presto-main/src/main/java/com/facebook/presto/metadata/BuiltInTypeAndFunctionNamespaceManager.java

```diff
@@ -1181,6 +1182,19 @@ else if (function instanceof SqlInvokedFunction) {
                     sqlFunction.getVersion(),
                     sqlFunction.getComplexTypeFunctionDescriptor());
         }
+        else if (function instanceof ParametricScalar) {
+            ParametricScalar sqlFunction = (ParametricScalar) function;
+            return new FunctionMetadata(
+                    signature.getName(),
+                    signature.getArgumentTypes(),
+                    signature.getReturnType(),
+                    signature.getKind(),
+                    JAVA,
+                    function.isDeterministic(),
+                    function.isCalledOnNullInput(),
+                    sqlFunction.getComplexTypeFunctionDescriptor(),
+                    sqlFunction.getScalarHeader().getSignatureToScalarStatsHeadersMap());
```

Comment:
Instead of passing the whole Map as an argument to FunctionMetadata, can we not just lookup the `ScalarStatsHeader` to see if it exists for the `signature` and pass in an `Optional<ScalarStatsHeader` ?